src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -328,8 +328,12 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
             options.miner.blockGasLimit,
             initialAccounts
           );
-          blocks.earliest = blocks.latest =
-            await this.#blockBeingSavedPromise.then(({ block }) => block);
+          blocks.latest = await this.#blockBeingSavedPromise.then(
+            ({ block }) => block
+          );
+          if (!blocks.earliest) {
+            blocks.earliest = blocks.latest;
+          }
```

Comment:
```suggestion
          blocks.earliest = blocks.latest;
```

Reply:
Previously I took @davidmurdoch's comment above to mean that we shouldn't be _setting_ earliest here. I think what he was actually saying is that we should _always_ be setting earliest here. If we've had to create a genesis block, we don't have an earliest or a latest, so they should be equal upon initialization. This seems to be the right approach to me.

Reply:
Yes, that is what I was saying. Sorry I wasn't clear. When you checked the code coverage for this do remember seeing if the (implicit) `else` of the `if (!blocks.earliest) {` was hit (which is what I thought you were saying in your previous comment)? IIRC, it'll show a black `E` on the same line as the `if` statement.

Reply:
Always assigning here overwrites the value set in BlockManager.initialize (via updateTaggedBlocks). So if there is a value we must keep it.
Maybe there is a better solution, but I don't have the understanding of the whole code to implement it.

Reply:
Okay, I think I understand now. This check is relevant for the following case:

When we are forking and initialize the fork, we get the remote's "latest" block, or whatever number the user forked at, but we don't actually set that to `this.latest` (this is on purpose, read on). After the fork is initialized, we initialize the block manager. In this forking case, the block manager's `updateTaggedBlocks` function will now return a result for `earliest` because, thanks to this change, we now check the fallback for an earliest block. However, that function will _not_ set a value for `this.latest` yet because no block has been saved to the database. After the block manager is initialized, we'll get here and find that `latest` has not yet been set, but `earliest` has. Because `latest` has not been set, `initializeGenesisBlock` will be called, which uses the fallback information to mine one block _above_ the remote's latest (or the fork block number). So this case would yield the expected result - `earliest` is whatever the remote returns for fetching earliest, and `latest` is one block mined on top of the fork block number.

@davidmurdoch  does that all seem right to you?

If so, this check should remain as-is. Though maybe to your original point this could be moved outside of the `if(!latest)` check.